xgcm/test/test_integrate.py

```diff
@@ -0,0 +1,355 @@
+    )
+
+    u_c = xr.DataArray(
+        data_generator(), coords=[("xu", xu), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+
+    v_c = xr.DataArray(
+        data_generator(), coords=[("xt", xt), ("yu", yu), ("time", t), ("zt", zt)]
+    )
+
+    wt = xr.DataArray(
+        data_generator(), coords=[("xt", xt), ("yt", yt), ("time", t), ("zw", zw)]
+    )
+
+    # maybe also add some other combo of x,t y,t arrays....
+    timeseries = xr.DataArray(np.random.rand(len(t)), coords=[("time", t)])
+
+    # northeast distance
+    dx = 0.3
+    dy = 2
+    dz = 20
+
+    dx_ne = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.1, coords=[("xu", xu), ("yu", yu)]
+    )
+    dx_n = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.2, coords=[("xt", xt), ("yu", yu)]
+    )
+    dx_e = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.3, coords=[("xu", xu), ("yt", yt)]
+    )
+    dx_t = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dx - 0.4, coords=[("xt", xt), ("yt", yt)]
+    )
+
+    dy_ne = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.1, coords=[("xu", xu), ("yu", yu)]
+    )
+    dy_n = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.2, coords=[("xt", xt), ("yu", yu)]
+    )
+    dy_e = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.3, coords=[("xu", xu), ("yt", yt)]
+    )
+    dy_t = xr.DataArray(
+        np.ones([len(xt), len(yt)]) * dy + 0.4, coords=[("xt", xt), ("yt", yt)]
+    )
+
+    dz_t = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zt", zt)]
+    )
+    dz_w = xr.DataArray(
+        data_generator() * dz, coords=[("xt", xt), ("yt", yt), ("time", t), ("zw", zw)]
+    )
+
+    # Make sure the areas are not just the product of x and y distances
+    area_ne = (dx_ne * dy_ne) + 0.1
+    area_n = (dx_n * dy_n) + 0.2
+    area_e = (dx_e * dy_e) + 0.3
+    area_t = (dx_t * dy_t) + 0.4
+
+    # calculate volumes, but again add small differences.
+    volume_t = (dx_t * dy_t * dz_t) + 0.25
+
+    def _add_metrics(obj):
+        obj = obj.copy()
+        for name, data in zip(
+            [
+                "dx_ne",
+                "dx_n",
+                "dx_e",
+                "dx_t",
+                "dy_ne",
+                "dy_n",
+                "dy_e",
+                "dy_t",
+                "dz_t",
+                "dz_w",
+                "area_ne",
+                "area_n",
+                "area_e",
+                "area_t",
+                "volume_t",
+            ],
+            [
+                dx_ne,
+                dx_n,
+                dx_e,
+                dx_t,
+                dy_ne,
+                dy_n,
+                dy_e,
+                dy_t,
+                dz_t,
+                dz_w,
+                area_ne,
+                area_n,
+                area_e,
+                area_t,
+                volume_t,
```

Comment:
I know this is how black wants it, but I can't stand this formatting. Is there some config we can tweak to change it?

Reply:
Their argument is that is produces nicer diffs. And there are no ways to tweak black, thats pretty much the idea behind it I think.
>Black reformats entire files in place. It is not configurable. It doesn't take previous formatting into account. It doesn't reformat blocks that start with # fmt: off and end with # fmt: on. # fmt: on/off have to be on the same level of indentation. It also recognizes YAPF's block comments to the same effect, as a courtesy for straddling code.

Is this such a concern that we should revert to another formatter? 
FWIW, I really like it since its enables easy commenting of input arguments, but I can see how it has a weird asthetic.

Reply:
Ok fine, no problem.

Reply:
For this very long list I could use a list of tuples directly (instead of zipping), which would cut this thing in half at least?